Week-3/exercise_3_1/index.js

```diff
@@ -0,0 +1,26 @@
+//Given reducer method:
+const add = (a, b = 0) => a + b;
+
+const memoize = (func) => {
+  // a cache of results
+  const results = {};
+  return (...args) => {
+    const argsKey = JSON.stringify(args);
+    if (!results[argsKey]) {
```

Comment:
@Saurabhrkp question will this search for property in prototype chain?

Reply:
is it required to seacrh

Reply:
yes it will search after it complete searching in properties